### PR TITLE
Improve codyze-cli + add output handling + add sarif output

### PR DESCRIPTION
**Pull request by maximilian-galanis**
Updates Codyze in three major ways:
1. introduces a new architecture for the CLI
2. Implements finding generation for Coko
3. Adds output handling to convert the internally used SARIF format to the chosen format (currently only SARIF output is implemented though)
4. Updates to a released cpg: v6.0.0
5. Updates to Java 17

New cli usage is now:
`./codyze --pedantic [other common options] runCoko --spec = ... [other coko specific options] cokoCpg --source=... [other cpg backend specific options]`

Each executor must now provide a `CliktCommand`, which is then injected using Koin. This allows every executor to customize the available backends.
Because each backend now also provides a `CliktCommand`, backends now have the option to manage their CMD options.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [versions]
 kotlin = "1.8.0"
-cpg = "main-SNAPSHOT"   # revert to official releases when features become available
+cpg = "6.0.0"   # revert to official releases when features become available
```
Comment (fwendland) [inline gradle/libs.versions.toml:3]: You can remove the comment with the reminder.

File: codyze-core/src/main/kotlin/de/fraunhofer/aisec/codyze/core/executor/ExecutorConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * This interface should be implemented as a data class to contain all the configuration options of an [Executor].
+ * An [ExecutorConfiguration] should also contain the shared codyze options passed to [ExecutorCommand.getExecutor].
+ *
+ * When using Codyze as a CLI program, the [ExecutorCommand] is responsible to instantiate the
+ * respective [Executor]. However, to facilitate the usage of Codyze as a library, an [Executor] should have a
+ * configuration object and a [Backend] as its two constructor arguments.
+ */
+interface ExecutorConfiguration {
+    fun normalize(): ExecutorConfiguration
```
Comment (fwendland) [inline codyze-core/src/main/kotlin/de/fraunhofer/aisec/codyze/core/executor/ExecutorConfiguration.kt:29]: Can you add a short description for the function?

File: codyze-core/src/main/kotlin/de/fraunhofer/aisec/codyze/core/config/CommonHelperFunctions.kt [hunk 3]
```diff
@@ -26,15 +26,15 @@ import kotlin.streams.asSequence
...
-            Files.walk(path.normalize().toAbsolutePath()).asSequence().filter { it.isRegularFile() }
-        )
-    }
-    return allSources
-}
+fun combineSources(vararg sources: List<Path>) = sources.toList().flatten().flatMap { path ->
+    // it is necessary to make the paths absolute because this function is used to combine paths
+    // that might be relative to different paths (relative to config file path <-> relative to CWD)
+    Files.walk(path.normalize().toAbsolutePath()).asSequence().filter { it.isRegularFile() }
+}.toSet()
```
[the remark this answers is not in the document]
Comment (maximilian-galanis) [reply codyze-core/src/main/kotlin/de/fraunhofer/aisec/codyze/core/config/CommonHelperFunctions.kt:33]: we could add a 'toRealPath' before the 'normalize' and 'toAbsolutePath' methods. That would resolve all sym links AFAIK. What do you think?

[the remark this answers is not in the document]
Comment (fwendland) [reply codyze-core/src/main/kotlin/de/fraunhofer/aisec/codyze/core/config/CommonHelperFunctions.kt:33]: We're going to defer possible challenges with symlinks when an issue occurs.

**Review by fwendland — APPROVED** [review]

**Pull request merged**